src/tdastro/astro_utils/passbands.py

```diff
@@ -0,0 +1,220 @@
+
+        This is eq. 8 from "On the Choice of LSST Flux Units" (Ivezić et al.):
+
+        φb(λ) = Sb(λ)λ⁻¹ / ∫ Sb(λ)λ⁻¹ dλ
+
+        Parameters
+        ----------
+        transmission_table : np.ndarray
+            A 2D array with wavelengths (Angstrom) in the first column and transmission strengths in
+            the second.
+
+        Returns
+        -------
+        np.ndarray
+            An array of phi_b values for all wavelengths in the transmission table.
+        """
+        # No interpolation in this version
+        transmission_values = transmission_table[:, 1]
+        wavelengths_angstrom = transmission_table[:, 0]
+        # Calculate the numerator
+        numerators = transmission_values / wavelengths_angstrom
+        # Perform trapezoidal integration over the wavelengths to get the denominators
+        denominators = scipy.integrate.trapezoid(
+            transmission_values / wavelengths_angstrom, x=transmission_table[:, 0]
+        )
+        # Calculate phi_b for all wavelengths
+        return numerators / denominators
+
+    def calculate_normalized_system_response_tables(self) -> None:
+        """Calculate the normalized system response tables for all bands."""
+        for band_name in self.bands:
+            normalized_wavelengths_angstroms = self._phi_b(self.transmission_tables[band_name])
+            self.normalized_system_response_tables[band_name] = np.column_stack(
+                (self.transmission_tables[band_name][:, 0], normalized_wavelengths_angstroms)
+            )
+
+    def _get_in_band_flux(self, flux: np.ndarray, normalized_system_response_table: np.ndarray) -> float:
+        """Calculate the in-band flux for a given flux and normalized system response table.
+
+        Note
+        ----
+        This version requires the input fluxes to match the wavelength grid in the normalized system
+        response table. A spline model can handle this by using the band's wavelength column as the
+        input wavelengths--but, this will be inconvenient for other types of models.
+
+        Let's consider adding another version that interpolates the normalized system response table
+        to the wavelengths of the input fluxes.
+
+        Parameters
+        ----------
+        flux : np.ndarray
+            Array of flux values evaluated at the wavelengths in the normalized system response table.
+        normalized_system_response_table : np.ndarray
+            A 2D array with wavelengths (Angstrom) in the first column and normalized system response
+            values in the second.
+
+        Returns
+        -------
+        float
+            The in-band flux value.
+        """
+        passband_wavelengths = normalized_system_response_table[:, 0]
+        integrand = flux * normalized_system_response_table[:, 1]
+        return scipy.integrate.trapezoid(integrand, x=passband_wavelengths)
+
+    def get_all_in_band_fluxes(self, model: PhysicalModel, times) -> np.ndarray:
```

Comment:
Going back to the point about single passband class: this method would not exist in that approach, but it is from "fun to have" category.

Reply:
#61